### ISSUE TITLE
Windows command prompt doesn't have an ssh client by default

### ISSUE THREAD

**Issue by jordan-webdev** [Varying-Vagrant-Vagrants/VVV#887]
I'm unable to navigate to vvv.dev in my browser after using the vagrant up command. I'm using windows 10. I installed virtual box in c/program files/oracle/virtualbox and I'm not sure where vagrant is, but I followed all the defaults in the install wizard. I put the vagrant-local directory straight in C://. Any clues? I'm using git shell, but I'm not sure if I logged in first in the shell. Does that matter?

Here's some of the code the shell outputs after I do vagrant up

Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/trusty64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: Setting the name of the VM: vagrant-local
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Remote connection disconnect. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 4.3.36
    default: VirtualBox Version: 5.0

**Comment by jordan-webdev** [comment]
Update: I had to use the command prompt as an administrator. I think this should be mentioned in the specs for noobs like me using Windows 10. I also used git bash to clone the repository, but I don't think that made a difference. I also shut my firewall off.

I can now visit all of the default sites : ), but I can't do the vagrant halt command! my command prompt tells me I need an 'ssh' executable. I followed the guide to using putty, but I can't do the command there either. Where do I do vagrant halt?

**Comment by lkwdwrd** [comment]
The command prompt doesn't have an ssh client by default. If you installed
the vagrant-triggers plugin it tries to SSH in and run commands before
halting the machine. You could uninstall vagrant-triggers (not
recommended), try running git-bash as an admin and running halt from there,
or you could add Git Bash to your path as described here (
https://ttmm.io/tech/vagrant-windows/), which gives you the ssh executable
in your command prompt. Good luck :)

**Comment by jordan-webdev** [comment]
Hey, I tried reinstalling git and selecting the option to use git in the command prompt during installation,  but command prompt is still not allowing me to halt. It says 'ssh' executable not found in any directories in the %PATH% variable. Is an SSH client installed? Try installing Cygwin, MinGW or Git,.....

**Comment by lkwdwrd** [comment]
Out of curiosity, did you restart your command session? In the prompt you
should be able to run `ssh -v` and see a version number of some kind. I'm
not on my windows machine right now so I can't grab example output or give
more explicit instructions, but it is complaining because it doesn't see an
ssh executable. You can also check your PATH variable to see what's in it.
I've got directions for that in the fourth bullet point here:
https://lkwdwrd.com/minor-windows-things/

**Comment by jordan-webdev** [comment]
Thank you for your help, I appreciate it. I tried restarting the command prompt, and when I enter 'ssh-v' it says it is not recognized as an internal or external command, operable program or batch file. This is my path: C:\PROGRA~2\Oracle\VirtualBox; I'm thinking that 2 is not correct.

**Comment by jordan-webdev** [comment]
I noticed your next bullet mentioned to add a HOME environment variable. I did that, with a value of Jordan (my user name), but unfortunately no luck.

**Comment by lkwdwrd** [comment]
I'll try to grab some values for you to check when I get back home and have
a moment to check the values in my Windows machine. In the meantime, I'd
suggest opening git-bash as an administrator and trying `vagrant halt` from
there. Git bash will have the ssh client built into the bash prompt it
gives you. No need to mess with your PATH that way. Still, it's handy so I
would suggest getting it set up at some point.

I've also been a big fan of running ConEmu on my windows machine for all of
my command prompts. Not the most pretty thing in the world, but helps keep
things a lot more organized and switching between different types (cmd, git
bash, etc.) with and without admin privs is a breeze.

**Comment by jordan-webdev** [comment]
Wow, it worked from git bash, thanks! Now for the moment of truth- to restart my computer and run vagrant up again. I might look into ConEmu, but for now is it a good idea to just handle all vagrant up and vagrant halt commands from git bash?

**Comment by lkwdwrd** [comment]
Ok, on my windows machine. of note there are two PATH vars. One for your user, one for the system. I just pulled my system PATH var and it looks like this:
```
%SystemRoot%\system32;%SystemRoot%;%SystemRoot%\System32\Wbem;%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Git\cmd;C:\Program Files (x86)\Git\bin;%USERPROFILE%\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\tools\php;C:\ProgramData\ComposerSetup\bin;C:\Program Files (x86)\Skype\Phone\;C:\tools\nodejs\;C:\HashiCorp\Vagrant\bin
```

reference: http://d.pr/i/1aW79

Notice in particular the `C:\Program Files (x86)\Git\cmd` entry. Yours should look similar. The in CMD you should be able to type

```
ssh -v
```

and get usage info. reference: http://d.pr/i/n3Tw

**Comment by jordan-webdev** [comment]
Interesting. You have all of these paths, but all I have is "C:\PROGRA~2\Oracle\VirtualBox;". Do you think I should add "C:\Program Files (x86)\Git\cmd" directly after that semicolon?

**Comment by lkwdwrd** [comment]
You should double check you are looking at the right Path. You are probably
looking at your user's Path var, not the System one. Your users's path is
in the top section of that window, you system Path will be in the bottom
section. If you chose to install git into your path, it should be there. If
not you should verify the install location and add it. Either to the system
path or your user's path. Depends if you want it available to other users
or not.

On Sunday, May 1, 2016, jordan-webdev <notifications@github.com> wrote:

> Interesting. You have all of these paths, but all I have is
> "C:\PROGRA~2\Oracle\VirtualBox;". Do you think I should add "C:\Program
> Files (x86)\Git\cmd" directly after that semicolon?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Varying-Vagrant-Vagrants/VVV/issues/887#issuecomment-216087126>
>


-- 
Luke Woodward – Associate Director of Engineering
10up Inc. <http://10up.com> | luke.woodward@10up.com | @lkwdwrd

**Comment by jordan-webdev** [comment]
Yes, I apologize, I saw the system tab later and then edited my post. In the systems tab, I have the following path "C:\Program Files\Git\cmd", but I noticed yours is "C:\Program Files (x86)\Git\cmd", and was wondering if I should change mine to match yours. I do have an x86 folder in my C directory.

**Comment by lkwdwrd** [comment]
It's going to depend on where you installed Git. Mine was installed in the
x86 folder, it looks like yours was not. This really should not affect how
it operates, but if you are concerned, you could look in each and see where
your Git folder is located. Since Git *is* in your path, give the `ssh -v`
command another try (note the space between 'ssh' and '-v'. If that is
working, double check you've gotten your HOME variable set correctly. I
have it set as a system level environment variable (bottom section) for
HOME.  'HOME' is the variable name and %USERPROFILE% is the value.
Reference: http://d.pr/i/KOJo

Once that is done, you should be all set to run vagrant commands from
either git bash or command. Both need to run as administrator, mostly
because of the hosts updater. You can't modify the hosts file without admin
privs. Hope that helps!

**Comment by jordan-webdev** [comment]
Yes, Git is installed in the folder it specified. I hadded the HOME variable with "%USERPROFILE%" as its value. For some reason it still isn't working, but at this point I'm just glad I got it to work using git bash. It would be nice to have this working, but I appreciate all your time. 

It's so bizarre. Here are some images of my environment variables.

home variable:

![image](https://cloud.githubusercontent.com/assets/16829239/14961006/8e2a9310-1065-11e6-882d-5c8d69e53a83.png)

environment variables path edit tab:

![image](https://cloud.githubusercontent.com/assets/16829239/14961080/dd5d45ae-1065-11e6-9404-076b89f6ca05.png)

**Comment by Sphinxxxx** [comment]
Somewhere along the way, the path changed to `C:\Program Files\Git\usr\bin`, so try:

    set PATH=%PATH%;C:\Program Files\Git\usr\bin

http://stackoverflow.com/a/38470601/1869660

**Comment by jeremyfelt** [comment]
Good info in this ticket. :) I've converted it to a docs ticket so that we can include it in some more official Windows troubleshooting and preparedness.

**Comment by Mte90** [comment]
I think that as today a lot of things are changed but I don't know if this one can be closed.
I think that this features now are available on windows

**Comment by tomjn** [comment]
I'm closing this out, it's a very old ticket, and a lot has changed since then
